File: mobile/android/src/main/kotlin/com/otravelz/android/data/network/dto/ItineraryDtos.kt
package com.otravelz.android.data.network.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ItineraryPlanRequestDto(
    val days: Int,
    val interests: List<String> = emptyList(),
    val pace: String? = null,
    val start: String? = null,
    @SerialName("budget_transport_per_day") val budgetTransportPerDay: Double? = null,
    @SerialName("low_walking") val lowWalking: Boolean? = null,
    val vegetarian: Boolean? = null
)

@Serializable
data class PlaceSummaryDto(
    val id: String,
    val name: String,
    val category: String
)

@Serializable
data class TransportHopDto(
    @SerialName("from_stop_id") val fromStopId: String? = null,
    @SerialName("to_stop_id") val toStopId: String? = null,
    val mode: String? = null,
    @SerialName("route_number") val routeNumber: String? = null,
    @SerialName("duration_minutes") val durationMinutes: Int? = null,
    val fare: Double? = null
)

@Serializable
data class ItineraryStopDto(
    val sequence: Int,
    val place: PlaceSummaryDto,
    @SerialName("planned_arrival") val plannedArrival: String? = null,
    @SerialName("planned_departure") val plannedDeparture: String? = null
)

@Serializable
data class ItineraryDayDto(
    @SerialName("day_number") val dayNumber: Int,
    val date: String? = null,
    val stops: List<ItineraryStopDto> = emptyList(),
    val hops: List<TransportHopDto> = emptyList()
)

@Serializable
data class ItineraryResponseDto(
    @SerialName("itinerary_id") val itineraryId: String,
    val days: List<ItineraryDayDto> = emptyList(),
    val explanation: String
)

File: mobile/ios/OTravelz/Networking/DTO/ItineraryDTOs.swift
import Foundation

public struct ItineraryPlanRequestDTO: Codable, Sendable {
    public let days: Int
    public let interests: [String]?
    public let pace: String?
    public let start: String?

    public init(days: Int, interests: [String]? = nil, pace: String? = nil, start: String? = nil) {
        self.days = days
        self.interests = interests
        self.pace = pace
        self.start = start
    }
}

public struct PlaceSummaryDTO: Codable, Sendable {
    public let id: String
    public let name: String
    public let category: String
}

public struct TransportHopDTO: Codable, Sendable {
    public let fromStopId: String?
    public let toStopId: String?
    public let mode: String?
    public let routeNumber: String?
    public let durationMinutes: Int?

    enum CodingKeys: String, CodingKey {
        case fromStopId = "from_stop_id"
        case toStopId = "to_stop_id"
        case mode
        case routeNumber = "route_number"
        case durationMinutes = "duration_minutes"
    }
}

public struct ItineraryStopDTO: Codable, Sendable {
    public let sequence: Int
    public let place: PlaceSummaryDTO
    public let plannedArrival: String?
    public let plannedDeparture: String?

    enum CodingKeys: String, CodingKey {
        case sequence
        case place
        case plannedArrival = "planned_arrival"
        case plannedDeparture = "planned_departure"
    }
}

public struct ItineraryDayDTO: Codable, Sendable {
    public let dayNumber: Int
    public let stops: [ItineraryStopDTO]
    public let hops: [TransportHopDTO]

    enum CodingKeys: String, CodingKey {
        case dayNumber = "day_number"
        case stops
        case hops
    }
}

public struct ItineraryResponseDTO: Codable, Sendable {
    public let itineraryId: String
    public let days: [ItineraryDayDTO]
    public let explanation: String

    enum CodingKeys: String, CodingKey {
        case itineraryId = "itinerary_id"
        case days
        case explanation
    }
}
